tap_appstore/streams.py

```diff
@@ -1,72 +1,264 @@
+        th.Property("proceeds_reason", th.StringType),
+        th.Property("preserved_pricing", th.StringType),
+        th.Property("client", th.StringType),
+        th.Property("order_type", th.StringType)
+    ).to_dict()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
+    def download_data(self, start_date, api):
+        filters = {
+            'frequency': 'DAILY',
+            'reportType': 'SALES',
+            'reportSubType': 'SUMMARY',
+            'reportDate': start_date,
+            'version': '1_0',
+            'vendorNumber': self.config['vendor_number']
+        }
+        return api.download_sales_and_trends_reports(filters=filters)
 
-# TODO: Delete this is if not using json files for schema definition
-SCHEMAS_DIR = importlib_resources.files(__package__) / "schemas"
-# TODO: - Override `UsersStream` and `GroupsStream` with your own stream definition.
-#       - Copy-paste as many times as needed to create multiple stream types.
 
+class SubscriberReportStream(client.AppStoreStream):
+    name = "subscriber_reports"
+    replication_key = "_api_report_date"
+    schema = th.PropertiesList(
+        th.Property("_line_id", th.IntegerType),
+        th.Property("_time_extracted", th.StringType),
+        th.Property("_api_report_date", th.DateTimeType),
+        th.Property("event_date", th.DateTimeType),
+        th.Property("app_name", th.StringType),
+        th.Property("app_apple_id", th.IntegerType),
+        th.Property("subscription_name", th.StringType),
+        th.Property("subscription_apple_id", th.IntegerType),
+        th.Property("subscription_group_id", th.IntegerType),
+        th.Property("standard_subscription_duration", th.StringType),
+        th.Property("promotional_offer_name", th.StringType),
+        th.Property("promotional_offer_id", th.StringType),
+        th.Property("subscription_offer_type", th.StringType),
+        th.Property("subscription_offer_duration", th.StringType),
+        th.Property("marketing_opt_in_duration", th.StringType),
+        th.Property("customer_price", th.NumberType),
+        th.Property("customer_currency", th.StringType),
+        th.Property("developer_proceeds", th.NumberType),
+        th.Property("proceeds_currency", th.StringType),
+        th.Property("preserved_pricing", th.StringType),
+        th.Property("proceeds_reason", th.StringType),
+        th.Property("client", th.StringType),
+        th.Property("device", th.StringType),
+        th.Property("country", th.StringType),
+        th.Property("subscriber_id", th.StringType),
+        th.Property("subscriber_id_reset", th.StringType),
+        th.Property("refund", th.StringType),
+        th.Property("purchase_date", th.StringType),
+        th.Property("units", th.IntegerType)
+    ).to_dict()
 
-class UsersStream(AppStoreStream):
-    """Define custom stream."""
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
```suggestion
```

Reply:
Added this back in as we now have to explicitly cast `event_date` to DateTime as some rows for it contain strings like `'  '`.